File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let max = 0;
    let i = 0;
    let substring = new Set();
    
    for (let j= 0; j < s.length; j++ ){
        while (substring.has(s[j])){
            substring.delete(s[i]);
                i += 1;
        }
        substring.add(s[j]);
        max = Math.max(max, substring.size)
    }
  return max
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        high = i = 0
        substring = set();

        for j in range(len(s)):
            while s[j] in substring:
                substring.remove(s[i])
                i += 1
            substring.add(s[j]);
            high = max(high, len(substring))
        return high